src/traces/bar/defaults.js

```diff
@@ -33,7 +155,23 @@ module.exports = function supplyDefaults(traceIn, traceOut, defaultColor, layout
     coerce('base');
     coerce('offset');
     coerce('width');
-    coerce('text');
+
+    coerceArray('text', coerceString);
+
+    var textPosition = coerceArray('textposition', coerceEnumerated);
+
+    var hasBoth = Array.isArray(textPosition) || textPosition === 'auto',
+        hasInside = hasBoth || textPosition === 'inside',
+        hasOutside = hasBoth || textPosition === 'outside';
+    if(hasInside || hasOutside) {
+        var textFont = coerceArray('textfont', coerceFont, layout.font);
+        if(hasInside) {
+            coerceArray('insidetextfont', coerceFont, textFont, layout.font);
+        }
+        if(hasOutside) {
+            coerceArray('outsidetextfont', coerceFont, textFont, layout.font);
+        }
+    }
```

Comment:
@etpinard What are the reasons against making `Plots.supplyDefaults` scale with the data? Note that the calls to `nestedProperty` would still scale only with the number of attributes.

Having uncoerced data in arrays after running `Plots.supplyDefaults` means that we have to coerce this data in `Bar.plot`. This increases the number of places where untrusted user input can brake Plotly.

Another disadvantage of moving the code for array coercion from `Bar.supplyDefaults` into `Bar.plot` (instead of `Lib`)  is that pie traces wouldn't be able to reuse the code.